src/wellKnownDidConfiguration/index.ts

```diff
@@ -61,155 +51,269 @@ function checkOrigin(input: string) {
+      proof.proofValue = MULTIBASE_BASE58BTC_HEADER + base58Encode(signature)
+      return { ...credentialBody, proof }
+    }
+    default:
+      throw new Error(`unknown proof type ${proofType}`)
+  }
+}
 
-  const credential = Credential.fromClaim(claim)
+function checkIsDomainLinkageCredential(credential: DomainLinkageCredential): void {
+  if (
+    !(
+      credential['@context']?.[0] === DEFAULT_VERIFIABLECREDENTIAL_CONTEXT &&
+      credential['@context']?.[1] === DID_CONFIGURATION_CONTEXT
+    )
+  ) {
+    throw new Error(
+      `credential must include contexts ${[DEFAULT_VERIFIABLECREDENTIAL_CONTEXT, DID_CONFIGURATION_CONTEXT]}`
+    )
+  }
+  if (
+    !(
+      credential.type?.includes(DOMAIN_LINKAGE_CREDENTIAL_TYPE) &&
+      credential.type?.includes(DEFAULT_VERIFIABLECREDENTIAL_TYPE)
+    )
+  ) {
+    throw new Error(
+      `credential must have types ${DEFAULT_VERIFIABLECREDENTIAL_TYPE} & ${DOMAIN_LINKAGE_CREDENTIAL_TYPE}`
+    )
+  }
+  try {
+    Did.validateUri(credential.credentialSubject?.id, 'Did')
+  } catch {
+    throw new Error('credentialSubject.id must be present and must be a DID')
+  }
+  if (credential.issuer !== credential.credentialSubject.id) {
+    throw new Error('issuer and credentialSubject.id must be identical')
+  }
+  try {
+    checkOrigin(credential.credentialSubject?.origin)
+  } catch {
+    throw new Error('credentialSubject.origin must be present and must be a valid domain origin')
+  }
+  if (typeof credential.issuanceDate !== 'string' || typeof credential.expirationDate !== 'string') {
+    throw new Error('issuanceDate & expirationDate must be present and must be iso date-time strings')
+  }
+}
 
-  const presentation = await Credential.createPresentation({
-    credential,
-    signCallback,
+export function didConfigResourceFromCredentials(credentials: DomainLinkageCredential[]): DidConfigResource {
+  credentials.forEach(checkIsDomainLinkageCredential)
+  credentials.reduce((last, next) => {
+    if (last.credentialSubject.origin !== next.credentialSubject.origin) {
+      throw new Error('credentials should have the same origin property')
+    }
+    return last
   })
-
-  if (presentation.claimerSignature.keyUri !== `${document.uri}${assertionKey.id}`) {
-    throw new Error('The credential presentation needs to be signed with the assertionMethod key')
+  return {
+    '@context': DID_CONFIGURATION_CONTEXT,
+    linked_dids: credentials,
   }
-
-  return presentation
 }
 
-export const DOMAIN_LINKAGE_CREDENTIAL_TYPE = 'DomainLinkageCredential'
-
-export async function didConfigResourceFromCredential(
-  credential: ICredentialPresentation,
-  expirationDate: string = new Date(Date.now() + 1000 * 60 * 60 * 24 * 365 * 5).toISOString()
+export async function createDidConfigResource(
+  signer: SignerInterface,
+  origin: string,
+  did: DidUri,
+  {
+    expirationDate = new Date(Date.now() + 1000 * 60 * 60 * 24 * 365 * 5),
```

Comment:
Is this a default of a year?

Reply:
Does this have any purpose or is it for show? E.g. Does the chain reflect this?

Reply:
this solution is completely off-chain. but credential verification checks this. And it's 5 years by default, I didn't change this.